src/iface/ethernet.rs

```diff
@@ -55,27 +66,40 @@ pub struct Interface<'b, 'c, DeviceT: for<'d> Device<'d>> {
 /// the `device` mutably until they're used, which makes it impossible to call other
 /// methods on the `Interface` in this time (since its `device` field is borrowed
 /// exclusively). However, it is still possible to call methods on its `inner` field.
-struct InterfaceInner<'b, 'c> {
+struct InterfaceInner<'b, 'c, 'e> {
     neighbor_cache:         NeighborCache<'b>,
     ethernet_addr:          EthernetAddress,
     ip_addrs:               ManagedSlice<'c, IpCidr>,
     #[cfg(feature = "proto-ipv4")]
     ipv4_gateway:           Option<Ipv4Address>,
+    #[cfg(feature = "proto-ipv4")]
+    ipv4_mcast_groups:      ManagedMap<'e, Ipv4Address, ()>,
```

Comment:
multicast addresses are used in IPv6 as well. See: [MLDv2]. The first implementation may only store `IpAddress::Ipv4` variants in this map, but I think the map should contain `IpAddress` instead of `Ipv4Address`.

[MLDv2]: https://tools.ietf.org/html/rfc3810

Reply:
Unlike `ip_addrs` (for IPv4) you may join a number of of groups. I didn't want to waste 16 bytes per IPv4 address. My use-case is Embedded Rust.

Reply:
Yes, it may optimize space for use-cases that are IPv4 only, but for use cases that are IPv4 and IPv6 we then have two `ManagedMap`s one for IPv4 and the other for IPv6.

Reply:
Even in dual-stack mode we save 12 bytes per IPv4 multicast group. Do you think that is not worth it?

Reply:
I would really like to keep IPv4 and IPv6 mcast groups in separate maps.

Memory space efficiency shouldn't depend on disabling IPv6. Seen that way, doing that could even reduce IPv6 adoption.

Reply:
I would still prefer that we have one map, but I can be okay with separate maps and I don't think it should keep this PR from getting merged.